gobblin-core-base/src/main/java/gobblin/writer/BytesBoundedBatch.java

```diff
@@ -48,52 +47,52 @@ public boolean isTTLExpire() {
     return (System.currentTimeMillis() - creationTimestamp) >= ttlInMilliSeconds;
   }
 
-  private long getInternalSize(String record) {
-    return record.length() + this.OVERHEAD_SIZE_IN_BYTES;
+  private long getInternalSize(D record) {
+    return (record).toString().length() + this.OVERHEAD_SIZE_IN_BYTES;
```

Comment:
Should consider having a record type to support non-string encoding and also avoid the cost of composing the string to get the length.